ParcialProgramacion4/src/Empresa.java

```diff
@@ -58,4 +60,371 @@ private Habilidad buscarHabilidad(String nombre) {
+        Scanner scanner = new Scanner(System.in);
+        //ingresar las habilidades y los años de experiencia en cada una
+        
+        //crear hashtable local
+        Hashtable<Habilidad, Integer> habilidades;
+        habilidades = new Hashtable<Habilidad, Integer>();
+
+        //llenar la hashtable con las habilidades
+        String otra;
+
+        do {
+            System.out.println("Nombre habilidad: ");
+            String nombreHabilidad = scanner.nextLine();
+
+            //busco que la habilidad exista
+            Habilidad habilidad = this.buscarHabilidad(nombreHabilidad);
+
+            if(habilidad == null) {
+                System.out.println("Descripcion: ");
+                String descripcion = scanner.nextLine();
+
+                habilidad = new Habilidad(nombreHabilidad, descripcion);
+
+                this.agregarHabilidad(habilidad);
+            }
+
+            //ya tengo la habilidad, verifico si ya esta agregada en la hashtable
+            if(!habilidades.containsKey(habilidad)) {
+                //ya se que no esta, ahora pido los años de experiencia en ella
+                System.out.println("Años de experiencia en "+nombreHabilidad+": ");
+                int annosExperiencia = Integer.parseInt(scanner.nextLine()); //porque despues va un nextLine
+
+                //agrego a la hashtable
+                habilidades.put(habilidad, annosExperiencia);
+
+                System.out.println("Habilidad agregada!!!");
+
+            } else
+                System.out.println("La habilidad ya esta agregada en la lista");
+
+            System.out.println("Agregar otra habilidad: (SI/NO)");
+            otra = scanner.nextLine(); //nextLinea porque despues va otro nextLine para evitar que se coma el salto de pagina
+            
+            while (!otra.equalsIgnoreCase("SI") && !otra.equalsIgnoreCase("NO")) {
+                System.out.println("Ingrese una opcion valida, SI o NO");
+                System.out.println("Agregar otra habilidad: (SI/NO)");
+                otra = scanner.nextLine();
+            }
+        } while (otra.equalsIgnoreCase("SI"));
+
+        return habilidades;
+    }
+    
+    //AGREGAR HABILIDAD PARA CASO DE USO AGREGAR EMPLEADO
+    //overriding para solo agregar a la lista una habilidad ya definida, se usa en pedirListaHabilidades
+    //NO ES EL METODO DEL CASO DE USO AGREGAR HABILIDAD
+    public void agregarHabilidad(Habilidad habilidadNueva) {
+        this.habilidades.add(habilidadNueva);
+        System.out.println("Habilidad registrada en la lista general de la empresa");
+    }
+
+    
+
+
+
+
+
+    //METODOS DE BUSQUEDA
+    private Empleado buscarEmpleado(int legajo) {
+        int i = 0;
+
+        while(i<empleados.size() && !empleados.get(i).hasEmpleado(legajo))
+            i++;
+        
+        if(i<empleados.size())
+            return empleados.get(i);
+        else
+            return null;
```

Comment:
no hace falta el else. porque si entro en el if, ya retorno
```suggestion
        if(i<empleados.size())
            return empleados.get(i);
        return null;
```